### PR TITLE
API calculates address_space automatically

### PR DESCRIPTION
**Pull request by tamirkamara**
# PR for issue #771 

## What is being addressed

Address spaces for workspaces should be calculated automatically without the user having to provide it.

## How is this addressed

- API allows creating a workspace without providing address_space in the payload which will result in an auto-generated one. 
- Auto-generated address space is a /24 free one within the overall TRE_ADDRESS_SPACE (new "core" param)
- No change in E2E which are still using custom addresses

### CODE REVIEW

File: management_api_app/db/repositories/workspaces.py [hunk 1]
```diff
@@ -60,6 +66,12 @@ def create_workspace_item(self, workspace_input: WorkspaceInCreate) -> Workspace
 
         return workspace
 
+    def get_new_address_space(self, cidr_netmask: int = 24):
```
Comment (marrobi) [inline management_api_app/db/repositories/workspaces.py:69]: Can the 24 be an environment variable? Will be cases in the near future needing larger address spaces for a workspace.

**Review by marrobi — COMMENTED** [review]
Looking great. Thanks!

File: management_api_app/db/repositories/workspaces.py (hunk 1 above)
Comment (tamirkamara) [reply management_api_app/db/repositories/workspaces.py:69]: We have another task #773 which might give the result you're looking for, but we can also have the default size as an environment variable.
TBH, I don't understand the flow... Is the user expected to know the size required for a workspace?
@marrobi

Comment (marrobi) [reply management_api_app/db/repositories/workspaces.py:69]: #773 will cover it for now. There are a few different flows, I can forsee. 

1. TRE Admin wants to specify specify address space. (custom) 
2. Address space automatically configured based on size specified by azure admin. (env var? )
3. TRE Admin wants to select from a t-shirt size.

1 and 2 are the ones feel will be most useful initially.

Comment (tamirkamara) [reply management_api_app/db/repositories/workspaces.py]: Resolving this as it will be handled in another task.

**Review by eladiw — APPROVED** [review]
looks good

**Comment by tamirkamara** [comment]
@marrobi yes, I wanted to rename ADDRESS_SPACE to CORE_ADDRESS_SPACE, but that is used in so many different places so preferred to do that in a specific task #804.

**Pull request merged**